File: POO/TP2/Practica Java/EJ4/Materia.java
package EJ4;

public class Materia {
    private String nombre;
    private String codigo;

    public Materia(String nombre, String codigo) {
        this.nombre = nombre;
        this.codigo = codigo;
    }

    public Materia(String nombre) {
        this.nombre = nombre;
    }

    public String getNombre() {
        return this.nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getCodigo() {
        return this.codigo;
    }

    public void setCodigo(String codigo) {
        this.codigo = codigo;
    }

}

File: POO/TP2/Practica Python/EJ4/Materia.py
class Materia:
    __nombre = ''
    __codigo = ''
    
    def __init__(self, nombre, codigo):
        self.__nombre = nombre
        self.__codigo = codigo
    
    @classmethod
    def iniciar_con_nombre(cls, nombre):
        materia = cls.__new__(cls)
        materia.__nombre = nombre
        return materia
    
    def get_nombre(self):
        return self.__nombre
    
    def set_nombre(self, nombre):
        self.nombre = nombre
    
    def set_codigo(self, codigo):
        self.__codigo = codigo